packages/integrations/vercel/src/serverless/entrypoint.ts

```diff
@@ -1,35 +1,21 @@
 import type { SSRManifest } from 'astro';
-import { App } from 'astro/app';
-import { applyPolyfills } from 'astro/app/node';
+import { applyPolyfills, NodeApp } from 'astro/app/node';
```

Comment:
Question: I've not been following the refactor changes, but with this change, it looks like it'll depend on https://github.com/withastro/astro/pull/9579. So does this mean that it's a breaking change for `@astrojs/vercel` that it relies on a new version of Astro to work?

Reply:
We are progressively pushing these refactors to the branch `adapter-enhancements`, eventually all the changes in the adapter will rely on the PR you linked. Eventually, all changes will be shipped together. However, I don't think `@astrojs/vercel` will be a major

Reply:
But won't it only work after that PR is released, aka the latest Astro version released then? An older version of Astro would not work with this new `@astrojs/vercel` changes?

Reply:
You're right. `@astronjs/vercel` will result in a `major` release.

cc @lilnasy

Reply:
That is correct. Peer dependency range will need to be updated to astro 4.2+.

Reply:
Cool. If we're also doing similar things for other adapters, maybe we can group it with this `@astrojs/node` pending major: https://github.com/withastro/astro/pull/9545 (if it's relevant).

Reply:
That makes sense to me. Do you think it would be easier if #9545 pointed to `adapter-enhancements`. It will be merged into main for astro 4.2.

Reply:
Yup I don't mind changing the base. If it makes it easier for you to make the changes later, feel free to merge that PR into your branch.